src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/ReadThreadingAssemblerArgumentCollection.java

```diff
@@ -166,6 +166,17 @@
     @Argument(fullName="graph-output", shortName="graph", doc="Write debug assembly graph information to this file", optional = true)
     public String graphOutput = null;
 
+    @Hidden
+    @Argument(fullName="debug-haplotype-discovery", doc="Write debug histogram for haplotype discovery", optional = true)
+    public boolean debugHaplotypeDiscovery = false;
+
+    /**
+     * This argument is meant for debugging and is not immediately useful for normal analysis use.
+     */
+    @Hidden
+    @Argument(fullName="haplotype-debug-histogram-output", doc="Write debug assembly graph information to this file", optional = true)
+    public String haplotypeHistogramOutput = null;
```

Comment:
Why not merge these two arguments into one? Eg., instead of a boolean and a file, just have the arg for the file, which if specified should activate the debut output, making the boolean superfluous. This is how several existing args like -bamout work.

Reply:
That seems fine, I was following the current paradigm for printing of the assembly graphs which uses a similar pattern. You are right that two arguments seems cumbersome. I will make that switch.